### ISSUE TITLE
Ensure that two chunks can be compared for equality and hash code

### ISSUE THREAD

**Issue by jdegoes** [zio/zio#1159]
`Chunk#equals` and `Chunk#hashCode` should observe the following properties for chunks `c1: Chunk[A]`, `c2: Chunk[A]`, and `c3: Chunk[A]`:

1. **Transitivity**. If `c1 == c2` and `c2 == c3`, then `c1 == c3`.
2. **Symmetry**. If `c1 == c2`, then `c2 == c1`.
3. **Reflexivity**. `c1 == c1`.
3. **Negation**. `c1 != c2 == !(c1 == c2)`
4. **Substitutivity**. If `c1 == c2` then `f(c1) == f(c2)` for all functions `f`.
5. **Hash Consistency**. If `c1 == c2`, then `c1.hashCode == c2.hashCode`.

We need to add tests for all these properties in some test cases in `ChunkSpec.scala`.

For example, here is one test of transitivity:

```scala
def testTransitivity = {
  val c1 = Chunk(1, 2, 3)
  val c2 = Chunk(1, 2, 3)
  val c3 = Chunk(1, 2, 3)
  ((c1 == c2) && (c2 == c3) && (c1 == c3)) must beTrue
```

Other tests can be added in similar fashion by following the coding conventions in `ChunkSpec.scala`.

In addition, we should probably ensure that `Chunk#toString`

**Comment by Vilkina** [comment]
I'll take it

**Comment by Vilkina** [comment]
Sorry, I don't know how to add 'Scala' code here. So, something like:
<img width="959" alt="Screen Shot 2019-07-10 at 9 56 43 PM" src="https://user-images.githubusercontent.com/3358827/60997597-da3a8680-a35f-11e9-83d3-0fb6f871b0d9.png">

**Comment by iravid** [comment]
Resolved in #1163
